engine/src/main/java/io/zeebe/engine/processor/workflow/handlers/CatchEventSubscriber.java

```diff
@@ -35,9 +35,10 @@ public CatchEventSubscriber(final CatchEventBehavior catchEventBehavior) {
       context.raiseIncident(ErrorType.EXTRACT_VALUE_ERROR, e.getMessage());
       return false;
     } catch (final MessageNameException e) {
-      // incident was already raised internally; processing was interrupted at the point the
-      // exception was thrown; no further processing shall take place, that's why we return false
-      // here
+      // processing was interrupted at the point the exception was thrown; no further processing
+      // shall take place, that's why we return false here
+      // todo verify that this is the correct level to raise the incident at
```

Comment:
Should we discuss this point? Or, is everything clear? :)

Reply:
It's not clear for me yet. The current implementation uses the 'normal' elementInstanceKey as scopeKey, but the MessageCorrelationKeyException is raising an incident at the `variablesScopeKey` level. Why is this is different?

Reply:
Good question. I guess that we should have the same logic for every expression. Maybe, we forgot something.